JSIL.Libraries/Includes/Bootstrap/Core/Classes/JSIL.StringEnumerator.js

```diff
@@ -0,0 +1,73 @@
+﻿JSIL.MakeClass("System.Object", "JSIL.StringEnumerator", true, ["T"], function ($) {
+  var T = new JSIL.GenericParameter("T", "JSIL.StringEnumerator");
+
+  $.RawMethod(false, "__CopyMembers__",
+    function ArrayEnumerator_CopyMembers(source, target) {
+      target._array = source._array;
+      target._length = source._length;
+      target._index = source._index;
+    }
+  );
+
+  $.Method({ Public: true, Static: false }, ".ctor",
+    new JSIL.MethodSignature(null, [$jsilcore.TypeRef("System.Array", ["!!0"]), $.Int32]),
+    function (array, startPosition) {
+      this._array = array;
+      this._length = array.length;
+      if (typeof (startPosition) !== "number")
+        JSIL.RuntimeError("ArrayEnumerator ctor second argument must be number");
+
+      this._index = startPosition;
+    }
+  );
+
+  $.Method({ Public: true, Static: false }, "Reset",
+    new JSIL.MethodSignature(null, []),
+    function () {
+      if (this._array === null)
+        JSIL.RuntimeError("Enumerator is disposed or not initialized");
+
+      this._index = -1;
+    }
+  );
+
+  $.Method({ Public: true, Static: false }, "MoveNext",
+    new JSIL.MethodSignature(System.Boolean, []),
+    function () {
+      return (++this._index < this._length);
+    }
+  );
+
+  $.Method({ Public: true, Static: false }, "Dispose",
+    new JSIL.MethodSignature(null, []),
+    function () {
+      this._array = null;
+      this._index = 0;
+      this._length = -1;
+    }
+  );
+
+  $.Method({ Public: false, Static: false }, null,
+    new JSIL.MethodSignature(System.Object, []),
+    function () {
+      return this._array.charCodeAt(this._index);
```

Comment:
I'll check it additionally, but really .Net strings (same as JS string) knows nothing about surrogate pairs.
So, it should just break them into two Chars (and Char is unable to represent surrogate pair, as it only Int16, and we need Int32 for surrogate pairs).